extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java

```diff
@@ -209,6 +211,11 @@ public void registerMetric(MetricRegistry.Type scope,
         }
     }
 
+    public void registerMetrics(Consumer<MetricsFactory> consumer) {
+        SmallRyeMetricsFactory factory = new SmallRyeMetricsFactory();
```

Comment:
Does it need a new factory for each consumer, or because each consumer creates a new builder from the factory, the factory can be the same?